pkg/transport/slowtransport.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package transport
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+)
+
+// SlowTransport is a slow version of transport
+type SlowTransport struct {
+	client  Client
+	latency time.Duration
+}
+
+// NewClientWithLatency makes a slower transport client for testing purposes
+func NewClientWithLatency(client Client, latency time.Duration) Client {
+	return &SlowTransport{
+		client:  client,
+		latency: latency,
+	}
+}
+
+// DialNode dials a node with latency
+func (slowTransport *SlowTransport) DialNode(ctx context.Context, node *pb.Node, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case <-time.After(slowTransport.latency):
+		break
+	}
+
+	return slowTransport.client.DialNode(ctx, node, opts...)
```

Comment:
I think we also want to implement slow read / write, not just the dialing part.

Reply:
we'll do this in another PR